testsuite/smoke-test/src/upgrade.rs

```diff
@@ -0,0 +1,160 @@
+    };
+
+    let update_gas_script = format!(
+        r#"
+    script {{
+        use aptos_framework::aptos_governance;
+        use aptos_framework::gas_schedule;
+
+        fun main(core_resources: &signer) {{
+            let framework_signer = aptos_governance::get_signer_testnet_only(core_resources, @0000000000000000000000000000000000000000000000000000000000000001);
+
+            let gas_bytes = {};
+
+            gas_schedule::set_gas_schedule(&framework_signer, gas_bytes);
+        }}
+    }}
+    "#,
+        generate_blob(&bcs::to_bytes(&gas_schedule).unwrap())
+    );
+
+    let gas_script_path = TempPath::new();
+    let mut gas_script_path = gas_script_path.path().to_path_buf();
+    gas_script_path.set_extension("move");
+    fs::write(gas_script_path.as_path(), update_gas_script).unwrap();
+
+    assert!(Command::new(aptos_cli.as_path())
+        .current_dir(workspace_root())
+        .args(&vec![
+            "move",
+            "run-script",
+            "--script-path",
+            gas_script_path.to_str().unwrap(),
+            "--sender-account",
+            "0xA550C18",
+            "--url",
+            url.as_str(),
+            "--private-key",
+            private_key.as_str(),
+            "--assume-yes",
+        ])
+        .output()
+        .unwrap()
+        .status
+        .success());
+
+    // Generate the governance proposal script.
+    let temp_script_path = TempPath::new();
+    let mut move_script_path = temp_script_path.path().to_path_buf();
+    move_script_path.set_extension("move");
+
+    let mut package_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf();
+    package_path.pop();
+    package_path.pop();
+    package_path.push("aptos-move/framework/aptos-framework");
+
+    assert!(Command::new(aptos_cli.as_path())
+        .current_dir(workspace_root())
+        .args(&vec![
+            "governance",
+            "generate-upgrade-proposal",
+            "--testnet",
+            "--account",
+            "0x1",
+            "--package-dir",
+            package_path.to_str().unwrap(),
+            "--output",
+            move_script_path.to_str().unwrap(),
+        ])
+        .output()
+        .unwrap()
+        .status
+        .success());
+
+    assert!(Command::new(aptos_cli.as_path())
+        .current_dir(workspace_root())
+        .args(&vec![
+            "move",
+            "run-script",
+            "--script-path",
+            move_script_path.to_str().unwrap(),
+            "--sender-account",
+            "0xA550C18",
+            "--url",
+            url.as_str(),
+            "--private-key",
+            private_key.as_str(),
+            "--assume-yes",
+        ])
+        .output()
+        .unwrap()
+        .status
+        .success());
+
+    // Two transactions has been emitted to root account.
+    *env.aptos_public_info().root_account().sequence_number_mut() += 2;
+
+    check_create_mint_transfer(&mut env).await;
```

Comment:
Let's also do module publishing here as well for additional testing post-upgrade